### PR TITLE
fix(ui5-checkbox): trigger form validation when required

### PR DESCRIPTION
**Pull request by kgogov**
The issue seems to happen because the internal `formSupport` slot has `disabled` attribute.
When inputs are disabled, the client does not send their values, resulting in the browser skipping validation.

Fixes: #7319

### CODE REVIEW

File: packages/main/src/CheckBox.ts [hunk 1]
```diff
@@ -321,7 +321,7 @@ class CheckBox extends UI5Element implements IFormElement {
 		const formSupport = getFeature<typeof FormSupport>("FormSupport");
 		if (formSupport) {
 			formSupport.syncNativeHiddenInput(this, (element: IFormElement, nativeInput: HTMLInputElement) => {
-				nativeInput.disabled = element.disabled || !element.checked;
+				nativeInput.disabled = element.disabled || (!element.required && !element.checked);
```
Comment (nnaydenow) [inline packages/main/src/CheckBox.ts:324]: I'm curious if we need that line at all. Do you know what would happen if we remove the line?

Comment (nnaydenow) [reply packages/main/src/CheckBox.ts:324]: My bad. disabled property of the native input should be linked with the disabled property of the ui5-checkbox so: `nativeInput.disabled = element.disabled || false` seems to be more correct

Comment (kgogov) [reply packages/main/src/CheckBox.ts]: Hi @nnaydenow,

Thank you very much for the feedback.
I made the necessary changes.

Comment (nnaydenow) [reply packages/main/src/CheckBox.ts]: Sorry again, but we I saw that checbkox should also provide checked information when validating an required attribute. Let's also add `nativeInput.checked = !!element.checked`

File: packages/main/test/pages/CheckBox.html [hunk 2]
```diff
@@ -83,6 +94,11 @@
 				event.preventDefault();
 			});
 		});
-</script>
+
+		cbForm.addEventListener("submit", function(event) {
+			event.preventDefault();
+			alert("Form submitted!");
```
Comment (nnaydenow) [inline packages/main/test/pages/CheckBox.html:100]: Minor: maybe we can enhance this handler a little bit and to add test about this functionality.

Comment (nnaydenow) [reply packages/main/test/pages/CheckBox.html:100]: You can check formElement.checkValidity() (this is the method that does the native validation of the forms). You can also use it for the test

**Review by PetyaMarkovaBogdanova — APPROVED** [review]

**Comment by kgogov** [comment]
Messed up the pull request, will open another.

**Pull request closed without merging**